src/wayland/idle_inhibit/mod.rs

```diff
@@ -0,0 +1,147 @@
+use wayland_protocols::wp::idle_inhibit::zv1::server as _idle_inhibit;
+use wayland_server::backend::GlobalId;
+use wayland_server::protocol::wl_surface::WlSurface;
+use wayland_server::{Client, DataInit, Dispatch, DisplayHandle, GlobalDispatch, New};
+
+use crate::wayland::idle_inhibit::inhibitor::IdleInhibitorState;
+
+pub mod inhibitor;
+
+const MANAGER_VERSION: u32 = 1;
+
+/// State of the zwp_idle_inhibit_manager_v1 Global
+#[derive(Debug)]
+pub struct IdleInhibitManagerState {
+    global: GlobalId,
+}
+
+impl IdleInhibitManagerState {
+    /// Create new [`zwp_idle_inhibit_manager`](wayland_protocols::idle_inhibit::zv1::server::zwp_idle_inhibit_manager_v1) global.
+    pub fn new<D>(display: &DisplayHandle) -> Self
+    where
+        D: GlobalDispatch<ZwpIdleInhibitManagerV1, ()>,
+        D: Dispatch<ZwpIdleInhibitManagerV1, ()>,
+        D: Dispatch<ZwpIdleInhibitorV1, IdleInhibitorState>,
+        D: IdleInhibitHandler,
+        D: 'static,
+    {
+        let global = display.create_global::<D, ZwpIdleInhibitManagerV1, _>(MANAGER_VERSION, ());
+
+        Self { global }
+    }
+
+    /// Returns the fractional scale manager global.
+    pub fn global(&self) -> GlobalId {
+        self.global.clone()
+    }
+}
+
+impl<D> GlobalDispatch<ZwpIdleInhibitManagerV1, (), D> for IdleInhibitManagerState
+where
+    D: GlobalDispatch<ZwpIdleInhibitManagerV1, ()>,
+    D: Dispatch<ZwpIdleInhibitManagerV1, ()>,
+    D: Dispatch<ZwpIdleInhibitorV1, IdleInhibitorState>,
+    D: IdleInhibitHandler,
+    D: 'static,
+{
+    fn bind(
+        _state: &mut D,
+        _display: &DisplayHandle,
+        _client: &Client,
+        manager: New<ZwpIdleInhibitManagerV1>,
+        _manager_state: &(),
+        data_init: &mut DataInit<'_, D>,
+    ) {
+        data_init.init(manager, ());
+    }
+}
+
+impl<D> Dispatch<ZwpIdleInhibitManagerV1, (), D> for IdleInhibitManagerState
+where
+    D: GlobalDispatch<ZwpIdleInhibitManagerV1, ()>,
+    D: Dispatch<ZwpIdleInhibitManagerV1, ()>,
+    D: Dispatch<ZwpIdleInhibitorV1, IdleInhibitorState>,
+    D: IdleInhibitHandler,
+    D: 'static,
+{
+    fn request(
+        state: &mut D,
+        _client: &Client,
+        _manager: &ZwpIdleInhibitManagerV1,
+        request: Request,
+        _data: &(),
+        _display: &DisplayHandle,
+        data_init: &mut DataInit<'_, D>,
+    ) {
+        match request {
+            Request::CreateInhibitor { id, surface } => {
+                state.inhibit(surface.clone());
+                data_init.init(id, IdleInhibitorState::new(surface));
+            }
+            Request::Destroy => (),
+            _ => unreachable!(),
+        }
+    }
+}
+
+/// Handler trait for idle-inhibit.
+pub trait IdleInhibitHandler {
+    /// Enable idle inhibition for the output of the provided surface.
+    fn inhibit(&mut self, surface: WlSurface);
+
+    /// Stop inhibition for the provided surface.
+    fn uninhibit(&mut self, surface: WlSurface);
```

Comment:
Could you add a comment describing, that this function is only called, if the client explicitly destroys the protocol object? Or maybe in simpler terms, that it is not called, if the surface is destroyed, even if it had an active inhibitor.